nemo/collections/nlp/modules/common/megatron/transformer.py

```diff
@@ -557,6 +587,9 @@ def forward(
         # change view [b, np, sq, hn]
         context_layer = context_layer.view(*output_size)
 
+        if self.headscale:
+            context_layer = context_layer * self.head_scale_tensor
+
```

Comment:
Missing a `LN` after the `HeadScaleMHA` ?
![image](https://user-images.githubusercontent.com/43824965/165166086-6edca000-6bad-457f-865c-284853e70fba.png)

Reply:
It happens outside the ParallelAttention layer in ParallelTransformerLayer_.